ts/features/idpay/payment/screens/IDPayPaymentResultScreen.tsx

```diff
@@ -86,6 +48,86 @@ const IDPayPaymentResultScreen = () => {
     </SafeAreaView>
   );
 };
+
+const SuccessScreenComponent = () => (
+  <View style={styles.resultScreenContainer}>
+    <Pictogram name={"completed"} size={120} />
+    <VSpacer size={24} />
+    <NewH4 style={styles.justifyTextCenter}>
+      {I18n.t("idpay.payment.result.success.title")}
+    </NewH4>
+    <VSpacer size={16} />
+    <Body style={styles.justifyTextCenter}>
+      {I18n.t("idpay.payment.result.success.body")}
+    </Body>
+  </View>
+);
+
+type FailureScreenContent = {
+  title: string;
+  subtitle?: string;
+  icon: IOPictograms;
+};
+
+const genericErrorContent: FailureScreenContent = {
+  title: I18n.t("idpay.payment.result.failure.GENERIC.title"),
+  subtitle: I18n.t("idpay.payment.result.failure.GENERIC.body"),
+  icon: "umbrella"
+};
+
+const failureScreenContent: Partial<
+  Record<PaymentFailure, FailureScreenContent>
+> = {
+  [PaymentFailureEnum.GENERIC]: genericErrorContent,
```

Comment:
why replicate the generic error content when you could just refer to it as `failureScreenContent[GENERIC]`?

Reply:
Because `Partial<Record<PaymentFailure, FailureScreenContent>>` is a `Partial` and accessing it with `failureScreenContent[GENERIC]` returns a `FailureScreenContent | undefined`. 
Forcing it with an `as FailureScreenContent` could fix this problem but I'd like to avoid using `as` unless strictly necessary.